eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,111 @@ function GetPackageLookup($packageList) {
   return $packageLookup
 }
 
+function create-metadata-table($readmePath, $moniker, $msService, $clientTableLink, $mgmtTableLink, $clientExists, $mgmtExists, $serviceName)
+{
+  New-Item -Path $readmePath -Force
+  $lang = $LanguageDisplayName
+  $langTitle = "Azure $serviceName SDK for $lang"
+  $header = GenerateDocsMsMetadata -language $lang -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Add-Content -Path $readmePath -Value $header
+
+  # Add tables, seperate client and mgmt.
+  $readmeHeader = "# $langTitle - $moniker"
+  Add-Content -Path $readmePath -Value $readmeHeader
+  if ($clientExists) {
+    $clientTable = "## Client packages - $moniker`r`n"
+    $clientTable += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $clientTable
+  }
+  if ($mgmtExists) {
+    $mgmtTable = "## Management packages - $moniker`r`n"
+    $mgmtTable += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $mgmtTable -NoNewline
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($readmePath, $serviceName, $msService)
+{
+  $readmeContent = Get-Content -Path $readmePath -Raw
+  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $metadataTable = $Matches["metadata"]
+  $metadataTable = ConvertFrom-StringData -StringData $metadataTable -Delimiter ':'
+  if (!$metadataTable) {
+    return
+  }
+  $restContent = $Matches["content"]
+  $serviceBaseName = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  $author = GetPrimaryCodeOwner -TargetDirectory "/sdk/$serviceBaseName/"
+  if (!$author) {
+    LogError "Cannot fetch the author from CODEOWNER file."
+    exit 1
+  }
+  $metadataTable["author"] = $author
+  $msauthor = GetMsAliasFromGithub -TenantId $tenantId -ClientId $clientId -ClientSecret $clientSecret -GithubUser $author   
+  if (!$msauthor) {
+    LogError "No ms.author found for $author. "
+    $msauthor = $author
+  }
+  $metadataTable["ms.author"] = $msauthor
+  $metadataTable["ms.service"] =  $msService
+  $metadataString = ""
+  $metadataTable.Keys | ForEach-Object {$metadataString += ($_ + ": " + $metadataTable[$_] + "`r`n")}
```

Comment:
Why not call GenerateDocsMsMetadata here as well so things are consistent?

Reply:
We just update the author, ms.author and ms.service instead of all metadata. The method of GenerateDocsMsMetadata update all. 

We can also update all metadata every time after we have the service level readme file.

Reply:
Yes it would be good to have only one place where we are building this string.